src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -184,12 +185,16 @@ private DetailAST getEventAst(TreeWalkerAuditEvent event) {
         DetailAST curNode = ast;
         while (curNode != null) {
             DetailAST toVisit = curNode.getFirstChild();
-            if (toVisit != null) {
-                childAstLineNoSet.add(toVisit.getLineNo());
-            }
+            addChildAstLineNo(childAstLineNoSet, toVisit);
             while (curNode != null && toVisit == null) {
                 toVisit = curNode.getNextSibling();
                 curNode = curNode.getParent();
+                if (curNode != null && curNode.equals(ast.getParent())) {
+                    break;
+                }
```

Comment:
You can move this condition to `while`. You also can use equals operators, because all these variables are references to nodes in the same AST tree:
```java
while (curNode != null && curNode != ast.getParent() && toVisit == null) {
```

Reply:
@baratali done, consider #162 has merged, these code moved